### DISCUSSION TITLE
question on Inboxes API

### DISCUSSION THREAD

**Discussion by dearkafka** [chatwoot/chatwoot#6689] [Chatwoot APIs]
Both path for "delete agent" and body has "inbox_id" param. why? Does it mean the same thing? can it be set to different values? whats the logic?

https://www.chatwoot.com/developers/api/#tag/Inboxes/operation/delete-agent-in-inbox

**Comment by dearkafka** [comment]
anyone?

**Comment by glauberssoutodossantos** [comment]
Hi @dearkafka, how are you?

I didn't understand exactly what you asked. But looking at the documentation there are two methods to delete an agent.

1) completely deletes the agent from the account.
https://www.chatwoot.com/developers/api/#tag/Agents/operation/delete-agent-from-account

2) removes agent permission from a specific inbox, but not from the account.
https://www.chatwoot.com/developers/api/#tag/Inboxes/operation/delete-agent-in-inbox

**Reply by dearkafka** [reply]
Thanks, Im fine! Trying to build complete chatwoot api wrapper in python. So, the question is why remove agent from inbox has 2 inbox_id, one in path and one in body? 

What I mean its 1) confusing 2) does not look like a good practice as it's difficult to understand difference or it is redundant

**Reply by glauberssoutodossantos** [reply]
Now I get your point. 

It really is quite redundant actually and unnecessary. I believe it was done that way, as it is the way to navigate the chatwoot frontend. The URL always shows the id of the account and the id of the accessed parameter (inbox, teams, agent, macro...).

I don't particularly like this form of navigation, but it's how the chatwoot team structured the project in the beginning.
